server/src/main/java/org/apache/iotdb/db/mpp/execution/executor/RegionWriteExecutor.java

```diff
@@ -112,17 +116,21 @@ public static ConsensusWriteResponse fireTriggerAndInsert(
       boolean hasFailedTriggerBeforeInsertion =
           result.equals(TriggerFireResult.FAILED_NO_TERMINATION);
 
+      planNode.setMetricTime(System.nanoTime());
       writeResponse = DataRegionConsensusImpl.getInstance().write(groupId, planNode);
 
       // fire Trigger after the insertion
       if (writeResponse.isSuccessful()) {
+        startTime = System.nanoTime();
         result = visitor.process(planNode, TriggerEvent.AFTER_INSERT);
         if (hasFailedTriggerBeforeInsertion || !result.equals(TriggerFireResult.SUCCESS)) {
           TSStatus triggerError = new TSStatus(TSStatusCode.TRIGGER_FIRE_ERROR.getStatusCode());
           triggerError.setMessage(
               "Meet trigger error before/after the insertion, the insertion itself is completed.");
           writeResponse = ConsensusWriteResponse.newBuilder().setStatus(triggerError).build();
         }
+        PerformanceOverviewMetricsManager.getInstance()
```

Comment:
Maybe we need to sum up the time of the trigger before and after phases as a statistic of how long a trigger takes?